build-tools/create-packs/Microsoft.Android.Sdk.proj

```diff
@@ -135,8 +137,8 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
         TargetFramework="$(_AndroidNETAppTargetFramework)"
         RuntimeFrameworkName="Microsoft.Android"
         LatestRuntimeFrameworkVersion="%24(_AndroidRuntimePackVersion)"
-        TargetingPackName="Microsoft.Android.Ref.%24(_AndroidRuntimePackId)"
-        TargetingPackVersion="%24(_AndroidRuntimePackVersion)"
+        TargetingPackName="Microsoft.Android.Ref.%24(_AndroidTargetingPackId)"
+        TargetingPackVersion="%24(_AndroidTargetingPackVersion)"
         RuntimePackNamePatterns="Microsoft.Android.Runtime.%24(_AndroidRuntimePackId).**RID**"
```

Comment:
I'm somewhat concerned about the approach to compile and run against different API levels. This reminds me of a very old issue we ran into when updating the shared runtime to always include the latest Mono.Android.dll - https://github.com/xamarin/androidtools/commit/0b7bcd77f1519d2928d9c8e619610ee2afe44786.

Perhaps that particular point of failure is no longer an issue in API 31+, as I think default interface methods in Java may have helped with this case? I can't think of any other scenarios that might break with this approach, but maybe @jonpryor has some additional thoughts here.

Reply:
Historical context: xamarin/androidtools@0b7bcd7 fixes [Xamarin Bugzilla 22074](https://bugzilla.xamarin.com/22/22074/bug.html), in which a `System.TypeLoadException` was thrown from a Debug-configuration app when the shared runtime was used, and the app was built with a `$(TargetFrameworkVersion)` "sufficiently lower than" the `$(TargetFrameworkVersion)` included in the shared runtime, all because of `android.database.Cursor`, my favorite type ever.  (Google kept adding methods to `Cursor`, even before Java interface default methods were supported, and it was a perfectly acceptable change…to *Java*, but not C#.  See also [API Design: Interfaces versus Abstract Classes](https://web.archive.org/web/20150911124932/blogs.oracle.com/darcy/entry/api_design_interfaces_versus_abstract).)

xamarin/androidtools@0b7bcd7 (2014-Aug) predates `FixAbstractMethodsStep` in xamarin/monodroid@73579e210e (2015-Nov); see also f96fcf93e157472072576bcc0a8698302899e8cf, 0c4c03303cacd377ddc536601e53e9006b7e4c2e, etc.  If `FixAbstractMethodStep` had existed, we could have instead "fixed up" the to-be-fast-deployed assemblies against the latest API level, which would also have addressed the scenario in xamarin/androidtools@0b7bcd7 while continuing to include the latest `Mono.Android.dll` in the shared runtime package.  xamarin/androidtools@0b7bcd7 also predates C#8 default interface members.

*Today*, with .NET 6, this *shouldn't* be a concern: we can now rely on C#8 default interface members, thus preserving API and ABI compatibility from lower API levels through higher API levels -- which means we could possibly consider *dropping* `FixAbstractMethodsStep` in .NET 6? -- and preserving API and ABI compatibility going forward is now one of our tenants.